lib/puppet/util/execution.rb

```diff
@@ -187,18 +187,30 @@ def self.execute(command, options = NoOptionsSpecified)
     null_file = Puppet.features.microsoft_windows? ? 'NUL' : '/dev/null'
 
     begin
+      reader, writer = IO.pipe unless options[:squelch]
```

Comment:
I don't think this is behavior people depend on it Puppet, so I modified the pxp-agent test to not expect it in https://github.com/puppetlabs/pxp-agent/pull/583.